share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,809 @@
+```
+def diff n m = match n m 
+  Nil x = x
+  x Nil = x
+  (_,t)(_,t1) = diff t t1
+```
+
+* Concantenates the given string with "ed"
+```
+def past1 n = "{n}ed"
+```
+
+* Multiplies every element in a list by 2
+```
+def multiply n = n | map (\x 2*x)
+```
+
+* Returns the element in the list of the given index n
+```
+def index n l = l | drop n | head
+```
+
+* Takes first n elements from the given list
+```
+def takeN n m = m | take n
+```
+
+* Drops first n elements from the given list
+```
+def dropN n m = m | drop n
+```
+
+* Creates a list with n as its first element and m as the last element 
+```
+def upto n m = m-n+1 | seq | map (\x x+n)
+```
+
+* Adds a space after a given string     
+```
+def addspace s = "{s} "
+```
+
+* Convert the String separated by spaces  into list of string 
+```
+def lex s = s | tokenize ` `
+```
+
+* Doubles every element in the list
+```
+def doublist n = match n 
+  Nil = Nil
+  h,t = 2*h , doublist t
+```
+
+* Increment the given list elements by 1
+```
+def inclist n = match n
+  Nil = Nil
+  h,t = (h+1), inclist t
+```
+
+* Doubles every element in the list using map function
+```
+def doublistmap n = n | map (\n 2*n)
+```
+
+* Increment every element in the list using map function
+```
+def inclistmap n = n | map (\n n+1) 
+```
+
+* Returns a list with the leading spaces removed
+```
+def dropspace n = match n
+  Nil = Nil
+  h,t = if (h ==* " ") then dropspace t else (h,t)
+```
+* Returns just the leading spaces
+```
+def takespace n = match n 
+  Nil = Nil
+  h,t = if (h ==* " ") then h,takespace t else Nil
+```
+
+* Concantenates every list of string with "io"
+```
+def map1 n = n | map (\s "{s}io")
+```
+
+* Creates a list (list Integers) from a given list of integers
+```
+def map2 n = n | map (\x x,Nil)
+```
+
+* Outputs the first element from the list (list Integers)
+```
+def map3 n = n | map (\x x | head | getOrElse 0)
+```
+
+* Creates a list of strings by talking the last character of a string
+```
+def map4 n = n | map (\x last x)
+```
+
+* Multiplies every element of the list by 3
+```
+def ftrl n = n | map (\x 3*x)
+```
+
+* Takes the first character from the list of strings and creates a new list
+```
+def fhel n = n | map (\x first x)
+```
+
+* removes first 2 characters from the list of strings and creates a new list
+```
+def tailstring n = n | explode | drop 1 |cat
+def fttl n = n | map (\x tailstring x)
+```
+
+* Prepend and Appends every element in a list of string by "s" "m"
+```
+def fsml n = n | map (\x "s{x}m")
+```
+
+### Creating your own Data types
+* Example shows how to create a data type of our own
+```
+global data Direction =
+  North
+  South
+  East
+  West
+```
+
+* Example shows the data type Money which can be in the form of Cash or the Cheque.
+```
+global data Money =
+  Cash Integer
+  Cheque (Pair String Double)
+```
+```
+def amount  = Cash 50
+def chequeamount = Cheque (Pair "HDFC" 350.00)
+```
+
+* Try out these and record the output
```

Comment:
what does this have to do with "creating your own data types"

Reply:
I have restructured that part